src/core/ext/xds/xds_transport.h

```diff
@@ -70,14 +69,45 @@ class XdsTransportFactory : public InternallyRefCounted<XdsTransportFactory> {
     virtual void ResetBackoff() = 0;
   };
 
+  // An interface for reporting connectivity state events for a transport.
+  //
+  // A transport can be in one of three states:
+  // - CONNECTING: transport is attempting to establish a connection
+  // - READY: transport is connected and ready to start streams
+  // - TRANSIENT_FAILURE: connection attempt has failed
+  //
+  // A transport is assumed to start in state CONNECTING.  Expected
```

Comment:
I think this comment should state that the transitions are written keeping pick first in mind

Reply:
I don't think that would really be appropriate, since this transport interface is supposed to abstract away gRPC-specific details like pick_first.  (It's probably already a bit more detailed than it needs to be, but I was trying to make it clear how to implement the interface for other systems internally.)